app/src/main/java/de/westnordost/streetcomplete/quests/construction/MarkCompletedHighwayConstruction.java

```diff
@@ -0,0 +1,106 @@
+package de.westnordost.streetcomplete.quests.construction;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.text.TextUtils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import de.westnordost.osmapi.map.data.BoundingBox;
+import de.westnordost.streetcomplete.R;
+import de.westnordost.streetcomplete.data.meta.OsmTaggings;
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder;
+import de.westnordost.streetcomplete.data.osm.download.MapDataWithGeometryHandler;
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao;
+import de.westnordost.streetcomplete.data.osm.tql.OverpassQLUtil;
+import de.westnordost.streetcomplete.quests.DateUtil;
+import de.westnordost.streetcomplete.quests.YesNoQuestAnswerFragment;
+
+public class MarkCompletedHighwayConstruction extends MarkCompletedConstruction
+{
+	@Inject
+	public MarkCompletedHighwayConstruction(OverpassMapDataDao overpassServer) {
+		super(overpassServer);
+	}
+
+	@Override public boolean download(BoundingBox bbox, MapDataWithGeometryHandler handler)
+	{
+		return overpassServer.getAndHandleQuota(getOverpassQuery(bbox), handler);
+	}
+
+	/** @return overpass query string to get streets marked as under construction but excluding ones
+	 * - with invalid construction tag
+	 * - with tagged opening date that is in future
+	 * - recently edited (includes adding/updating check_date tags)
+	 * - with fixme (as in some situations this quest may add fixme note as an edit)
+	 * . */
+	private String getOverpassQuery(BoundingBox bbox)
+	{
+		String groupName = ".roads_under_construction";
+		String resultsName = ".roads_for_review";
+		return OverpassQLUtil.getGlobalOverpassBBox(bbox) +
+			"way" + getQueryPart("highway", groupName, 14) +
+			"(" +
+			"  way[highway=construction][fixme];" +
+			") -> .with_fixme;" +
```

Comment:
I see no reason to now show `highway=construction` with fixme. Do you?

Reply:
Because of the fixme tag added by this quest? https://github.com/westnordost/StreetComplete/pull/920#discussion-diff-176043863R63

Reply:
This quest may add fixme as one of resulting of edits.

Discarding objects with fixme hides objects where this quest is already solved.

fixme may be instead appended to existing one if present and this check may check whatever fixme string contains text added by this quest as a substring - bit I think that it is not worth aging this complexity/

Reply:
But it also adds `highway=...` and removes the `highway=construction` tagging, even in the fixme case. So, after the user gave an answer, the quest will not turn up for other users anyway.

Reply:
Oh, it doesn't, there is a return. But why? It could mark it as construction completed but add the fixme?

Reply:
I though that without changing highway=construction to highway=* situation will be more clear to the next editor - who will be more likely to understands why this access tags were added.